modules/videoio/src/cap_android_camera.cpp

```diff
@@ -478,29 +515,31 @@ class AndroidCameraCapture : public IVideoCapture
                     }
                 }
             case CAP_PROP_AUTO_EXPOSURE:
-                autoExposure = (value != 0);
+                aeMode = (value != 0) ? ACAMERA_CONTROL_AE_MODE_ON : ACAMERA_CONTROL_AE_MODE_OFF;
                 if (isOpened()) {
-                    uint8_t aeMode = autoExposure ? ACAMERA_CONTROL_AE_MODE_ON : ACAMERA_CONTROL_AE_MODE_OFF;
-                    camera_status_t status = ACaptureRequest_setEntry_u8(captureRequest.get(), ACAMERA_CONTROL_AE_MODE, 1, &aeMode);
-                    return status == ACAMERA_OK;
+                    return submitRequest(ACaptureRequest_setEntry_u8, ACAMERA_CONTROL_AE_MODE, aeMode);
                 }
                 return true;
             case CAP_PROP_EXPOSURE:
                 if (isOpened() && exposureRange.Supported()) {
-                    exposureTime = (int64_t)value;
+                    exposureTime = exposureRange.clamp(static_cast<int64_t>(value));
                     LOGI("Setting CAP_PROP_EXPOSURE will have no effect unless CAP_PROP_AUTO_EXPOSURE is off");
-                    camera_status_t status = ACaptureRequest_setEntry_i64(captureRequest.get(), ACAMERA_SENSOR_EXPOSURE_TIME, 1, &exposureTime);
-                    return status == ACAMERA_OK;
+                    return submitRequest(ACaptureRequest_setEntry_i64, ACAMERA_SENSOR_EXPOSURE_TIME, exposureTime);
                 }
                 return false;
             case CAP_PROP_ISO_SPEED:
                 if (isOpened() && sensitivityRange.Supported()) {
-                    sensitivity = (int32_t)value;
+                    sensitivity = sensitivityRange.clamp(static_cast<int32_t>(value));
                     LOGI("Setting CAP_PROP_ISO_SPEED will have no effect unless CAP_PROP_AUTO_EXPOSURE is off");
-                    camera_status_t status = ACaptureRequest_setEntry_i32(captureRequest.get(), ACAMERA_SENSOR_SENSITIVITY, 1, &sensitivity);
-                    return status == ACAMERA_OK;
+                    return submitRequest(ACaptureRequest_setEntry_i32, ACAMERA_SENSOR_SENSITIVITY, sensitivity);
                 }
                 return false;
+            case CAP_PROP_ANDROID_DEVICE_TORCH:
+                flashMode = (value != 0) ? ACAMERA_FLASH_MODE_TORCH : ACAMERA_FLASH_MODE_OFF;
```

Comment:
There are complains about Samsung devices support here: https://stackoverflow.com/questions/23336907/what-is-the-difference-between-flash-mode-torch-and-flash-mode-on. Could you take a look?

Reply:
Hmm, I don't know. My primary test device is a SONY. I will try, if FLASH_MODE_ON will work as well.

Reply:
Just tested torch with Xiaomi Mi 10 and Samsung Galaxy S10. Both works well. Let's ignore it for now.

Reply:
According to https://developer.android.com/reference/android/hardware/Camera.Parameters  FLASH_MODE_ON is part of the old API. Camera2 has FLASH_MODE_OFF, FLASH_MODE_SINGLE and FLASH_MODE_TORCH only.